src/main/java/core/basesyntax/service/implementations/CarServiceImpl.java

```diff
@@ -0,0 +1,72 @@
+package core.basesyntax.service.implementations;
+
+import core.basesyntax.dao.DaoCar;
+import core.basesyntax.lib.Inject;
+import core.basesyntax.lib.Service;
+import core.basesyntax.model.Car;
+import core.basesyntax.model.Driver;
+import core.basesyntax.service.CarService;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Service
+public class CarServiceImpl implements CarService {
+    @Inject
+    private DaoCar dao;
+
+    @Override
+    public Car create(Car item) {
+        return dao.create(item);
+    }
+
+    @Override
+    public Car get(Long id) {
+        return dao.get(id).get();
+    }
+
+    @Override
+    public List<Car> getAll() {
+        return dao.getAll();
+    }
+
+    @Override
+    public Car update(Car item) {
+        return dao.update(item);
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        return dao.delete(id);
+    }
+
+    @Override
+    public void addDriverToCar(Driver driver, Car car) {
+        List<Driver> drivers = car.getDrivers();
+        drivers.add(driver);
+        car.setDrivers(drivers);
+        dao.update(car);
+    }
+
+    @Override
+    public void removeDriverFromCar(Driver driver, Car car) {
+        if (car.getDrivers().contains(driver)) {
```

Comment:
real db will handle this check and throw exception if no driver is present, so let's remove it